saleor/graphql/payment/mutations.py

```diff
@@ -134,22 +134,32 @@ def clean_payment_amount(cls, info, checkout_total, amount):
             )
 
     @classmethod
-    def validate_gateway(cls, manager, gateway_id, currency):
+    def validate_gateway(cls, manager, gateway_id, checkout):
         """Validate if given gateway can be used for this checkout.
 
-        Check if provided gateway_id is on the list of available payment gateways.
-        Gateway will be rejected if gateway_id is invalid or a gateway doesn't support
-        checkout's currency.
+        Check if provided gateway is active and CONFIGURATION_PER_CHANNEL is True.
+        If CONFIGURATION_PER_CHANNEL is False then check if gateway has
+        defined currency.
         """
-        if not is_currency_supported(currency, gateway_id, manager):
-            raise ValidationError(
-                {
-                    "gateway": ValidationError(
-                        f"The gateway {gateway_id} is not available for this checkout.",
-                        code=PaymentErrorCode.NOT_SUPPORTED_GATEWAY.value,
-                    )
-                }
-            )
+        payment_gateway = manager.get_plugin(gateway_id, checkout.channel.slug)
+
+        if not payment_gateway or not payment_gateway.active:
+            cls.raise_not_supported_gateway_error(gateway_id)
+
+        if not payment_gateway.CONFIGURATION_PER_CHANNEL:
```

Comment:
WOuldn't be good to pass this check responsibility to the plugins? There is no need to force Saleor, to know, what Plugin's class value means. I mean, we have `is_currency_supported`, the base implementation in BasePlugin, could check this value, and return expected response OR error